src/main/webapp/js/finscholar/homepagecontroller/homepagecontroller.js

```diff
@@ -22,48 +22,88 @@ const JsactionEventContract = goog.require('jsaction.EventContract');
 const GoogDom = goog.require('goog.dom');
 const GoogSoy = goog.require('goog.soy');
 const {homepage} = goog.require('finscholar.homepagecontroller.templates');
+const {CollegeListView} = goog.require('finscholar.collegelistview');
+const {ScholarshipListView} = goog.require('finscholar.scholarshiplistview');
 const {PageController} = goog.require('pagecontroller');
 
 /**
  * Class for the home page controller.
  * @public
  */
 class HomePageController extends PageController {
-  constructor() {
+
+  /** 
+   * @constructor
+   * @param {!Element} container The HTML div where the main frame is rendered.
+   */
+  constructor(container) {
     super();
+    /** @private @const @type {!Element} */
+    this.container_ = container;
+    /** @private @const @type {!JsactionEventContract} */
     this.eventContract_ = new JsactionEventContract();
+    /** @private @const @type {!JsactionDispatcher} */
+    this.dispatcher_ = new JsactionDispatcher();
+    /** @private @const bindedNavbarOnclickHandler_ The binded navbar onclick event handler. */
+    this.bindedNavbarOnclickHandler_ = this.handleNavbarOnclickEvent_.bind(this);
+    this.initJsaction_();
+    /** @private */
```

Comment:
If not a const, you can do something like `/** @private @type {!number} */`.

Reply:
Thanks. I just added js doc for all member variables. And integers are nun-nullable by default